src/utils/saveFileHandler.ts

```diff
@@ -24,61 +26,63 @@ export interface SaveData {
     };
 }
 
-const CURRENT_VERSION = 1;
+export interface PPK2Metadata {
+    metadata: {
+        samplesPerSecond: number;
+        recordingDuration: number;
+    };
+}
+const CURRENT_VERSION = 2;
 
-const write = (deflateRaw: DeflateRaw) => (data: unknown) =>
-    new Promise(resolve => {
-        deflateRaw.write(data, 'binary', resolve);
+export default async (
+    filename: string,
+    metadata: PPK2Metadata,
+    sessionFolder: string,
+    onProgress: (message: string) => void
+) => {
+    await startPreventSleep();
+    const metaPath = path.join(sessionFolder, 'metadata.json');
+    DataManager().saveMinimap(sessionFolder);
+    fs.writeFileSync(
+        metaPath,
+        JSON.stringify({ ...metadata, formatVersion: CURRENT_VERSION })
+    );
+
+    const output = fs.createWriteStream(filename);
+    const archive = archiver('zip', {
+        zlib: { level: 6 }, // Sets the compression level.
```

Comment:
might be worth ectracting to the config / .env?

Reply:
might add these config to the UI bit not for now ...